### PR TITLE
Reuse `CMA` inside `CMAwM`

### PR DESCRIPTION
**Pull request by knshnb**
`CMAwM` has much in common with `CMA` in logic.
This PR simplifies the implementation of `CMAwM` by letting `CMAwM` have `CMA` as an attribute and reusing its logic.

### CODE REVIEW

**Comment by knshnb** [comment]
Let me work on the refactoring of the `CMAwM` logic itself by another PR.

**Review by c-bata — APPROVED** [review]
LGTM!

@nomuramasahir0 Please merge this PR after you approved.

**Comment by nomuramasahir0** [comment]
@knshnb @c-bata 
Thank you for the nice simplification! LGTM!

FYI: We may want to create a unique `should_stop` function for `CMAwM`, as the current `should_stop` for `CMA` function is designed for optimization over continuous variables. However, to do so, some considerations and numerical investigations are necessary, and I believe the current implementation is sufficient for the time being.

**Pull request merged**
